src/overlays/actors/ovl_En_Test7/z_en_test7.c

```diff
@@ -32,80 +46,943 @@ const ActorInit En_Test7_InitVars = {
+        if (temp_v0 != NULL) {
+            gSPMatrix(POLY_OPA_DISP++, temp_v0, G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+            gSPDisplayList(POLY_OPA_DISP++, gameplay_keep_DL_081628);
+        }
+    }
+
+    if (ptr) {}
+
+    Matrix_StatePop();
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
+
+void func_80AF1730(EnTest7Struct* arg0) {
+    arg0->unk_00 = 0.0f;
+    arg0->unk_04 = 0.0f;
+    arg0->unk_08 = 1.0f;
+    arg0->unk_0C = 1.0f;
+    arg0->unk_10 = 0;
+}
+
+void EnTest7_Init(Actor* thisx, GlobalContext* globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+    EnTest7* this = THIS;
+    Player* player = GET_PLAYER(globalCtx);
+    Player* player2 = GET_PLAYER(globalCtx);
+
+    this->actor.world.rot.y = this->actor.shape.rot.y = player->actor.shape.rot.y;
+    this->unk_144 = 0;
+    this->unk_1E54 = 0;
+    this->unk_1E8E = player->actor.shape.rot.y;
+    this->unk_1E90 = player->actor.scale.x;
+    this->unk_1E94 = player->actor.scale.z;
+
+    func_80183430(&this->unk_18CC, &gameplay_keep_Blob_085640, &gameplay_keep_Blob_083534, this->unk_18FC,
+                  this->unk_1BA8, 0);
+    func_801834A8(&this->unk_18CC, &gameplay_keep_Blob_083534);
+    func_80AF0838(this->unk_15C);
+    func_80AF1730(&this->unk_148);
+
+    if (ENTEST7_GET(&this->actor) == ENTEST7_MINUS1) {
+        func_80AF082C(this, func_80AF2938);
+        EnTest7_SetupAction(this, NULL);
+    } else {
+        func_80AF082C(this, func_80AF19A8);
+        EnTest7_SetupAction(this, func_80AF2854);
+        func_801A2E54(NA_BGM_SONG_OF_SOARING);
+    }
+
+    if (globalCtx->playerActorCsIds[8] == -1) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+
+    ActorCutscene_SetIntentToPlay(globalCtx->playerActorCsIds[8]);
+    player2->stateFlags1 |= 0x20;
+    Lights_PointNoGlowSetInfo(&this->lightInfo, (Math_SinS(this->unk_1E8E) * 90.0f) + player->actor.world.pos.x,
+                              player->actor.world.pos.y + 10.0f,
+                              (Math_CosS(this->unk_1E8E) * 90.0f) + player->actor.world.pos.z, 255, 255, 255, 255);
+    this->lightNode = LightContext_InsertLight(globalCtx, &globalCtx->lightCtx, &this->lightInfo);
+}
+
+void EnTest7_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnTest7* this = THIS;
+
+    ActorCutscene_Stop(globalCtx->playerActorCsIds[8]);
+    LightContext_RemoveLight(globalCtx, &globalCtx->lightCtx, this->lightNode);
+}
+
+void func_80AF19A8(EnTest7* this, GlobalContext* globalCtx) {
+    if (!ActorCutscene_GetCanPlayNext(globalCtx->playerActorCsIds[8])) {
+        ActorCutscene_SetIntentToPlay(globalCtx->playerActorCsIds[8]);
+    } else {
+        ActorCutscene_Start(globalCtx->playerActorCsIds[8], NULL);
+        func_80AF082C(this, func_80AF1A2C);
+        globalCtx->unk_18844 = 1;
+    }
+}
+
+void func_80AF1A2C(EnTest7* this, GlobalContext* globalCtx) {
+    Color_RGB8 sp34 = D_80AF342C;
+    Color_RGB8 sp30 = D_80AF3430;
```

Comment:
```suggestion
    Color_RGB8 sp34 = D_80AF342C;
    Color_RGB8 sp30 = D_80AF3430;
```
Looks to just be declared on the stack.